lib/phoenix_token_auth/controllers/users_controller.ex

```diff
@@ -14,44 +14,39 @@ defmodule PhoenixTokenAuth.Controllers.Users do
   Params should be:
       {user: {email: "user@example.com", password: "secret"}}
 
-  If successfull, sends a welcome email.
+  To be considered valid, a record must pass both validations *and* constraints.
+  Validations can be run before querying the db, but we have to insert/update the
+  record to determine if all constraints have been fulfilled.
 
-  Responds with status 200 and body "ok" if successfull.
+  If successful, sends a welcome email.
+
+  Responds with status 200 and body "ok" if successful.
   Responds with status 422 and body {errors: {field: "message"}} otherwise.
   """
   def create(conn, params = %{"user" => %{"email" => email}}) when email != "" and email != nil do
     {confirmation_token, changeset} = Registrator.changeset(params["user"])
     |> Confirmator.confirmation_needed_changeset
 
-    if changeset.valid? do
-      case Util.repo.transaction fn ->
-        user = Util.repo.insert!(changeset)
-        Mailer.send_welcome_email(user, confirmation_token, conn)
-      end do
-        {:ok, _} -> json conn, :ok
-      end
+    if changeset |> RegistrationFlow.user_created_and_emailed?(confirmation_token, conn) do
+      conn |> json :ok
     else
-      Util.send_error(conn, Enum.into(changeset.errors, %{}))
+      conn |> respond_with_errors(changeset.errors)
```

Comment:
i also notices another problem: we won't get the "uniqueness errors" into the response this way. they will only get added in `RegistrationFlow.user_created...` and we never use the result from that.
i think we need to return `{:error, changeset}` from the `RegistrationFlow` methods and use that here.